orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/TerminateInstancesStage.groovy

```diff
@@ -14,32 +14,37 @@
  * limitations under the License.
  */
 
-package com.netflix.spinnaker.orca.kato.pipeline
+package com.netflix.spinnaker.orca.clouddriver.pipeline
 
-import groovy.transform.CompileStatic
+import com.netflix.spinnaker.orca.clouddriver.pipeline.support.TerminatingInstanceSupport
 import com.netflix.spinnaker.orca.clouddriver.tasks.MonitorKatoTask
-import com.netflix.spinnaker.orca.kato.tasks.TerminateInstancesTask
-import com.netflix.spinnaker.orca.kato.tasks.WaitForTerminatedInstancesTask
+import com.netflix.spinnaker.orca.clouddriver.tasks.TerminateInstancesTask
+import com.netflix.spinnaker.orca.clouddriver.tasks.WaitForTerminatedInstancesTask
 import com.netflix.spinnaker.orca.pipeline.LinearStage
 import com.netflix.spinnaker.orca.pipeline.model.Stage
 import org.springframework.batch.core.Step
+import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
 @Component
-@CompileStatic
 class TerminateInstancesStage extends LinearStage {
 
   public static final String PIPELINE_CONFIG_TYPE = "terminateInstances"
 
+  @Autowired
+  TerminatingInstanceSupport instanceSupport
+
   TerminateInstancesStage() {
     super(PIPELINE_CONFIG_TYPE)
   }
 
   @Override
   public List<Step> buildSteps(Stage stage) {
-    def step1 = buildStep(stage, "terminateInstances", TerminateInstancesTask)
-    def step2 = buildStep(stage, "monitorTermination", MonitorKatoTask)
-    def step3 = buildStep(stage, "waitForTerminatedInstances", WaitForTerminatedInstancesTask)
-    [step1, step2, step3]
+    stage.context[TerminatingInstanceSupport.TERMINATE_REMAINING_INSTANCES] = instanceSupport.remainingInstances(stage)
```

Comment:
I'm trying to think through why this should happen at StageBuilder time as opposed to as the first operation inside the TerminateInstancesTask with TerminateInstances producing a stageOutput

Feels like that is the more appropriate time to make this lookup given that it should be pulling the launchTime right before the instance is terminated

Also if it was performed inside the Task, then the TerminatingInstance should be possible to be fully populated (instanceId + launchTime) and Immutable?